otx/cli/manager/config_manager.py

```diff
@@ -0,0 +1,405 @@
+
+        Args:
+            subsets (list, str): Defaults to ["train", "val", "unlabeled"].
+
+        Returns:
+            dict: dataset_config
+        """
+        dataset_config = {"task_type": self.task_type}
+        for subset in subsets:
+            if f"{subset}_subset" in self.data_config and self.data_config[f"{subset}_subset"]["data_root"]:
+                dataset_config.update({f"{subset}_data_roots": self.data_config[f"{subset}_subset"]["data_root"]})
+        return dataset_config
+
+    def update_data_config(self, data_yaml: dict) -> None:
+        # TODO: This also requires uniformity in the format.
+        """Convert the data yaml format to the data_config format consumed by the task.
+
+        Args:
+            data_yaml (dict): data.yaml format
+        """
+        if data_yaml["data"]["train"]["data-roots"]:
+            self.data_config["train_subset"] = {"data_root": data_yaml["data"]["train"]["data-roots"]}
+        if data_yaml["data"]["val"]["data-roots"]:
+            self.data_config["val_subset"] = {"data_root": data_yaml["data"]["val"]["data-roots"]}
+        if data_yaml["data"]["test"]["data-roots"]:
+            self.data_config["test_subset"] = {"data_root": data_yaml["data"]["test"]["data-roots"]}
+        if "unlabeled" in data_yaml["data"] and data_yaml["data"]["unlabeled"]["data-roots"]:
+            self.data_config["unlabeled_subset"] = {
+                "data_root": data_yaml["data"]["unlabeled"]["data-roots"],
+                "file_list": data_yaml["data"]["unlabeled"]["file-list"],
+            }
+        # FIXME: Hardcoded for Self-Supervised Learning
+        if self.mode == "train" and str(self.train_type).upper() == "SELFSUPERVISED":
+            self.data_config["val_subset"] = {"data_root": None}
+
+    def _get_template(self, task_type: str, model: str = None) -> ModelTemplate:
+        """Returns the appropriate template for each situation.
+
+        Args:
+            task_type (str): The task_type registered in the registry. Used for filtering.
+            model (str, optional): The task_type registered in the registry. Used for filtering. Defaults to None.
+
+        Returns:
+            ModelTemplate: Selected model template.
+        """
+        otx_registry = OTXRegistry(self.otx_root).filter(task_type=task_type)
+        if model:
+            template_lst = [temp for temp in otx_registry.templates if temp.name.lower() == model.lower()]
+            if len(template_lst) == 0:
+                raise ValueError(
+                    f"[*] {model} is not a type supported by OTX {task_type}."
+                    f"\n[*] Please refer to 'otx find --template --task_type {task_type}'"
+                )
+            template = template_lst[0]
+        else:
+            template = otx_registry.get(DEFAULT_MODEL_TEMPLATE_ID[task_type.upper()])
+        return template
+
+    def build_workspace(self, new_workspace_path: str = None) -> None:
```

Comment:
```suggestion
    def build_workspace(self, new_workspace_path: Optional[Union[str, Path]] = None) -> None:
```

Reply:
Here new_workspace_path is intended as str. It is intended to be passed directly from args. I'll leave it like this.

Reply:
```suggestion
    def build_workspace(self, new_workspace_path: Optional[str] = None) -> None:
```
Then, how about this?

Reply:
reflected this. https://github.com/openvinotoolkit/training_extensions/pull/1641/commits/56c882a32f8d2476219990539bd17ef17903629e